### PR TITLE
ref(server): Refactors storage for manifests 

### PR DESCRIPTION
**Pull request by thomastaylor312**
This solved the problem where you could really only update one manifest at a time per lattice. Please note that this has some added e2e tests, but those are currently commented out as they are waiting on some of our other work for removing jitter from starting actors (which is landing in the new host version and in wadm fairly soon)

### CODE REVIEW

File: src/scaler/manager.rs [hunk 1]
```diff
@@ -198,11 +196,23 @@ where
             .map_err(|e| anyhow::anyhow!("Unable to subscribe to consumer: {e:?}"))?;
 
         // Get current scalers set up
-        let all_manifests = manifest_store.list::<StoredManifest>(lattice_id).await?;
+        let manifest_store = crate::server::ModelStorage::new(manifest_store);
+        let futs = manifest_store
```
Comment (brooksmtownsend) [inline src/scaler/manager.rs:200]: 🦶🏻

File: src/server/storage.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    #[instrument(level = "debug", skip(self))]
+    pub async fn list(&self, lattice_id: &str) -> Result<Vec<ModelSummary>> {
+        debug!("Fetching list of models from storage");
+        let futs = self
+            .get_model_set(lattice_id)
+            .await?
+            .unwrap_or_default()
+            .0
+            .into_iter()
+            // We can't use filter map with futures, but we can use map and then filter_map it out below
```
Comment (brooksmtownsend) [inline src/server/storage.rs:97]: Did you want to add this `filter_map` or was it not necessary in the end?

File: src/server/types.rs [hunk 3]
```diff
@@ -57,7 +57,7 @@ pub struct ModelSummary {
     pub name: String,
     pub version: String,
     pub description: Option<String>,
-    pub deployed: bool,
+    pub deployed_version: Option<String>,
```
Comment (brooksmtownsend) [inline src/server/types.rs:60]: :alert: breaking change, this will need to be updated in `wash` as well

**Review by brooksmtownsend — APPROVED** [review]

File: src/server/storage.rs (hunk 2 above)
Comment (thomastaylor312) [reply src/server/storage.rs:97]: It wasn't necessary. I'll remove the comment

File: src/server/types.rs (hunk 3 above)
Comment (thomastaylor312) [reply src/server/types.rs:60]: I think this is more accurate and useful though! Do you?

**Pull request merged**